sourcetracker/_sourcetracker.py

```diff
@@ -583,16 +634,109 @@ def gibbs_sampler(sink, cp, restarts, draws_per_restart, burnin, delay):
                 final_env_assignments[drawcount] = seq_env_assignments
                 final_taxon_assignments[drawcount] = taxon_sequence
 
-                # We've made a draw, update this index so that the next
-                # iteration will be placed in the correct index of results.
+                if autoc is not None:
+                    converged = autoc(final_envcounts, autoc_drawcount,
+                                      drawcount)
+                    if converged:
+                        convergence_times[restart, 0] = drawcount
+                        convergence_times[restart, 1] = autoc_drawcount
+                        break
+                    elif not converged and rep < total_passes:
+                        autoc_drawcount += 1
+                    elif not converged and rep == total_passes:
+                        convergence_times[restart, 0] = drawcount
+                        convergence_times[restart, 1] = -1
```

Comment:
Can `rep` be greater than `total_passes`? Do you need a case for that in this if/elif?